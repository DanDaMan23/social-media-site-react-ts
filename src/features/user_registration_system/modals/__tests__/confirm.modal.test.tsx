import { render, screen } from "@testing-library/react"
import { FC, useEffect, useState } from "react"
import ConfirmModal from "../confirm.modal"

interface IRenderConfirmModal {
  id?: string
  show: boolean
  onClose?: jest.Mock
  onSubmit?: jest.Mock
}

const renderConfirmModal = ({
  id = "mockId",
  show,
  onClose = jest.fn(),
  onSubmit = jest.fn()
}: IRenderConfirmModal) => {
  const TestComponent: FC = () => {
    const [showModal, setShowModal] = useState<boolean>(false)

    useEffect(() => {
      setShowModal(show)
    }, [])

    return (
      <ConfirmModal
        id={id}
        show={showModal}
        onClose={onClose}
        onSubmit={onSubmit}
      />
    )
  }

  return render(<TestComponent />)
}

describe("confirm.modal test", () => {
  it("should render modal when show is true", async () => {
    const { baseElement } = renderConfirmModal({ show: true })

    expect(baseElement).toMatchSnapshot()
  })

  it("should render modal when show is false", async () => {
    const { baseElement } = renderConfirmModal({ show: false })

    expect(baseElement).toMatchSnapshot()
  })
})
