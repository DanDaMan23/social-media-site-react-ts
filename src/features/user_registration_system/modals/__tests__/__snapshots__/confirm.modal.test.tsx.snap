// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm.modal test should render modal when show is false 1`] = `
<body
  class=""
  style="padding-right: 1024px;"
>
  <div />
</body>
`;

exports[`confirm.modal test should render modal when show is true 1`] = `
<body
  class="modal-open"
  data-rr-ui-modal-open=""
  style="padding-right: 1024px;"
>
  <div />
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      id="mockId"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
          >
            Confirm Registration
          </div>
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          <p>
            Are your provided information correct?
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            No
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
